pkg/backend/refresh.go

```diff
@@ -34,9 +52,10 @@ func (b *backend) refreshToken(ctx context.Context, storage logical.Storage, key
 		return nil, err
 	} else if tok == nil {
 		return nil, nil
-	} else if tok.Valid() || tok.RefreshToken == "" {
+	} else if tokenOk2Reuse(tok, data) || tok.RefreshToken == "" {
 		return tok, nil
 	}
+	tok.AccessToken = ""
```

Comment:
Won't this cause all tokens that don't use refreshing to stop working?

Reply:
It only gets to this point if the token is not valid anyway, so I don't see the harm in erasing it, do you?  It's about to attempt to be replaced via a refresh token and if that fails, an error is returned anyway.

Reply:
And actually, it already returns before that point whenever tok.RefreshToken is not set.

Reply:
Makes sense, thanks for the clarification!